src/server/scripts/EasternKingdoms/SunwellPlateau/boss_muru.cpp

```diff
@@ -110,53 +139,73 @@ class boss_entropius : public CreatureScript
 
         void Reset() override
         {
-            DoCastAOE(SPELL_NEGATIVE_ENERGY_E);
+            _Reset();
+            DoCast(me, SPELL_ENTROPIUS_COSMETIC_SPAWN, true);
         }
 
-        void EnterCombat(Unit* /*who*/) override
+        void ScheduleTasks() override
         {
-            _EnterCombat();
-            DoCastAOE(SPELL_NEGATIVE_ENERGY_E, true);
-            DoCast(me, SPELL_ENTROPIUS_SPAWN);
-            events.ScheduleEvent(EVENT_SUMMON_BLACK_HOLE, 15000);
+            scheduler.Schedule(Milliseconds(2000), [this](TaskContext /*context*/)
+            {
+                std::list<Creature*> portals;
+                me->GetCreatureListWithEntryInGrid(portals, NPC_MURU_PORTAL_TARGET, 100.0f);
+                for (std::list<Creature*>::iterator itr = portals.begin(); itr != portals.end(); ++itr)
+                    (*itr)->RemoveAllAuras();
+
+                DoCastAOE(SPELL_NEGATIVE_ENERGY_PERIODIC_E, true);
+            });
+
+            scheduler.Schedule(Seconds(15), [this](TaskContext context)
+            {
+                DoCastAOE(SPELL_DARKNESS_E, true);
+                DoCastAOE(SPELL_BLACKHOLE, true);
+
+                context.Repeat(Seconds(15));
```

Comment:
You can also do just `context.Repeat()` since the TaskContext know's it's last scheduled time.